From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Branson <andrew.branson@jolla.com>
Date: Wed, 5 Aug 2026 17:49:09 +0200
Subject: [sailfishos][download] Select the hosted helper app dialog

Do not register toolkit's desktop helper-app dialog for the Sailfish
application. The EmbedLite application component table supplies its own
contract, keeping the hosted picker out of generic toolkit code.
---
 toolkit/mozapps/downloads/components.conf | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/toolkit/mozapps/downloads/components.conf b/toolkit/mozapps/downloads/components.conf
index ccaf5e46c189..999e8af10f9e 100644
--- a/toolkit/mozapps/downloads/components.conf
+++ b/toolkit/mozapps/downloads/components.conf
@@ -2,11 +2,14 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-Classes = [
-    {
-        'cid': '{F68578EB-6EC2-4169-AE19-8C6243F0ABE1}',
-        'contract_ids': ['@mozilla.org/helperapplauncherdialog;1'],
-        'esModule': 'resource://gre/modules/HelperAppDlg.sys.mjs',
-        'constructor': 'nsUnknownContentTypeDialog',
-    },
-]
+Classes = []
+
+if buildconfig.substs['MOZ_BUILD_APP'] != 'mobile/sailfishos':
+    Classes += [
+        {
+            'cid': '{F68578EB-6EC2-4169-AE19-8C6243F0ABE1}',
+            'contract_ids': ['@mozilla.org/helperapplauncherdialog;1'],
+            'esModule': 'resource://gre/modules/HelperAppDlg.sys.mjs',
+            'constructor': 'nsUnknownContentTypeDialog',
+        },
+    ]
